js_methods.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/StringLiterals, Style/AndOr
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield to_a[i] if block_given?
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield to_a[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    temp_arr = []
+    my_each do |e|
+      temp = yield e
+      temp_arr << e if temp
+    end
+    temp_arr
+  end
+
+  def my_all?
+    ans = true
+    my_each do |e|
+      ans = false unless yield e
+    end
+    ans
+  end
+
+  def my_any?
+    ans = false
+    my_each do |e|
+      ans = true if yield e
+    end
+    ans
+  end
+
+  def my_none?
+    ans = true
+    my_each do |e|
+      ans = false if yield e
+    end
+    ans
+  end
```

Comment:
**[REQUIRED]**
* when no block or argument is given returns true only if none of the collection members is true.
* when a class is passed as an argument returns true if none of the collection is a member of such class.
* when a Regex is passed as an argument returns true only if none of the collection matches the Regex.
* when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.